archunit/src/main/java/com/tngtech/archunit/core/domain/AccessTarget.java

```diff
@@ -440,8 +440,9 @@ public JavaClass getRawReturnType() {
         @Override
         @PublicAPI(usage = ACCESS)
         public ThrowsClause<CodeUnitCallTarget> getThrowsClause() {
+            @SuppressWarnings("RedundantTypeArguments") // Some JDK versions need the type argument, or it will not compile
             List<ThrowsClause<JavaCodeUnit>> resolvedThrowsClauses = FluentIterable.from(resolve())
-                    .transform(toGuava(JavaCodeUnit.Functions.Get.throwsClause()))
+                    .transform(toGuava(JavaCodeUnit.Functions.Get.<JavaCodeUnit>throwsClause()))
```

Comment:
Ah this [tripped me up as well](https://github.com/TNG/ArchUnit/pull/724/commits/3dd094d200ec257f17380bc3d77cc902eb274c5e); nice to see another way to resolve it!